### PR TITLE
Add `tenant_uuid` and `user_uuids` to queue caller and queue member events

### PR DESCRIPTION
**Pull request by farhad2161**
Hi,

For events like `queue_member_status`, I extracted agent id from `Interface` property of bus event by using regex. Then I used confd client to find `tenant_uuid` of the agent.

Do you have any better suggestion?

### CODE REVIEW

**Comment by miladrazban** [comment]
@sboily Hi Sylvain

All the events both for Agents and Queues are working now. It's ready for your review.

File: wazo_calld_queue/plugin.py [hunk 1]
```diff
@@ -24,18 +25,24 @@ def load(self, dependencies):
         token_changed_subscribe = dependencies['token_changed_subscribe']
         bus_consumer = dependencies['bus_consumer']
         bus_publisher = dependencies['bus_publisher']
-
+        auth_client = AuthClient(**config['auth'])
+        token = auth_client.token.new(expiration=365 * 24 * 60 * 60)['token']
```
Comment (sboily) [inline wazo_calld_queue/plugin.py:29]: Why are you doing this here? I don't understand.

File: wazo_calld_queue/bus_consume.py [hunk 2]
```diff
@@ -105,7 +151,42 @@ def _queue_member_status(self, event):
...
+        headers = self._build_headers(
+            user_uuids=user_uuids,
+            tenant_uuid=tenant_uuid,
+        )
+        self.bus_publisher.publish(bus_event, headers=headers)
 
     def _publish(self, bus_event):
         self.bus_publisher.publish(bus_event, headers={'name': bus_event.name})
+
+    def _extract_tenant_from_event(self, event):
```
Comment (sboily) [inline wazo_calld_queue/bus_consume.py:164]: ouch, we need to think to a better solution, if it work for you it's great, this is clearly not possible to have to do that for building something on the top of Wazo Platform. Performance will also probably decrease with this function.

File: wazo_calld_queue/bus_consume.py [hunk 3]
```diff
@@ -3,17 +3,28 @@
 # SPDX-License-Identifier: GPL-3.0+
 
 import logging
-
+import re
```
Comment (sboily) [inline wazo_calld_queue/bus_consume.py:6]: clearly not a fan to use re for event system.

File: wazo_calld_queue/plugin.py [hunk 4]
```diff
@@ -24,18 +25,24 @@ def load(self, dependencies):
...
+        confd_client = ConfdClient(**config['confd'])
+        confd_client.set_token(token)
         amid_client = AmidClient(**config['amid'])
 
         token_changed_subscribe(amid_client.set_token)
 
-        queues_bus_event_handler = QueuesBusEventHandler(bus_publisher)
+        queues_bus_event_handler = QueuesBusEventHandler(bus_publisher, confd_client)
```
Comment (sboily) [inline wazo_calld_queue/plugin.py:36]: Clearly not a good idea for performance.

**Review by sboily — COMMENTED** [review]
Thank you for this PR, if this PR work for you it's good, but the workaround for the new event system in Wazo Platform is not good for performance. I will talk with people to find the better way to fix this part also for other modules. I just review and i don't think about the good solution.

**Comment by mwolff44** [comment]
Hi,
I tested your PR, i have this issue : 
`tail -f /var/log/wazo-calld.log
2022-10-05 14:06:36,771 [15918] (DEBUG) (wazo_calld_queue.bus_consume): {'Event': 'QueueMemberStatus', 'Privilege': 'agent,all', 'Queue': 'sales-ucaas', 'MemberName': 'Local/13aa6332-8bc1-4e36-bf5a-8b0c51d6ccff@usersharedlines', 'Interface': 'Local/13aa6332-8bc1-4e36-bf5a-8b0c51d6ccff@usersharedlines', 'StateInterface': 'hint:13aa6332-8bc1-4e36-bf5a-8b0c51d6ccff@usersharedlines', 'Membership': 'static', 'Penalty': '0', 'CallsTaken': '0', 'LastCall': '0', 'LastPause': '0', 'LoginTime': '1664978654', 'InCall': '0', 'Status': '1', 'Paused': '0', 'PausedReason': '', 'Ringinuse': '1', 'Wrapuptime': '0', 'Skills': ''}
2022-10-05 14:06:36,771 [15918] (ERROR) (CoreBusConsumer): Handler '_queue_member_status' for event 'QueueMemberStatus' failed
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/xivo_bus/mixins.py", line 201, in __dispatch
    handler(payload)
  File "/usr/local/lib/python3.7/dist-packages/wazo_calld_queue/bus_consume.py", line 154, in _queue_member_status
    tenant_uuid, user_uuids = self._extract_tenant_from_event(event)
  File "/usr/local/lib/python3.7/dist-packages/wazo_calld_queue/bus_consume.py", line 175, in _extract_tenant_from_event
    if len(agent_regex.groups()):
AttributeError: 'NoneType' object has no attribute 'groups`

regards

Mathias

**Comment by sboily** [comment]
It's because the regexp don't match your event. regexp_agent is empty.

**Comment by sboily** [comment]
This features has been included in other PR.

**Pull request closed without merging**